packages/backend/src/helpers/flow-templates.ts

```diff
@@ -52,32 +44,86 @@ async function createFlowFromTemplate(
     await flow.$relatedQuery('steps', trx).insert({
       type: 'trigger',
       position: 1,
-      appKey: triggerAppKey,
-      key: triggerEventKey,
+      appKey: steps[0].appKey,
+      key: steps[0].eventKey,
     })
 
     // insert action steps
-    for (let i = 0; i < actions.length; i++) {
-      const { appKey: actionAppKey, eventKey: actionEventKey } = actions[i]
+    for (let i = 1; i < steps.length; i++) {
+      const step: TemplateStep = steps[i]
       await flow.$relatedQuery('steps', trx).insert({
         type: 'action',
-        position: i + 2, // start at position 2
-        appKey: actionAppKey,
-        key: actionEventKey,
+        position: step.position,
+        appKey: step.appKey,
+        key: step.eventKey,
       })
     }
 
     logger.info('Demo flow created', {
       event: 'demo-flow-request',
       flowId: flow.id,
       flowName: flow.name,
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +38 to +65]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->can this be wrapped in a transaction, in case it fails halfway we wont have any half-created dangling pipes<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/716?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->